### PR TITLE
DREF & PER changes

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/views/PerWorkPlanForm/index.tsx [hunk 1]
```diff
@@ -263,26 +264,27 @@ export function Component() {
...
+    const readOnlyMode = isNotDefined(statusResponse)
+        || isNotDefined(statusResponse.phase)
+        || statusResponse.phase < PER_PHASE_WORKPLAN;
 
     return (
         <Container
             className={styles.perWorkPlanForm}
             childrenContainerClassName={styles.content}
-            footerActions={(
+            footerActions={value.is_draft !== false && (
```
Comment (tnagorra) [inline src/views/PerWorkPlanForm/index.tsx:275]: What is value.is_draft is undefined? Shouldn't it be treated as false?

File: src/views/PerPrioritizationForm/index.tsx [hunk 2]
```diff
@@ -416,15 +417,27 @@ export function Component() {
...
-                    <Button
-                        name={undefined}
-                        variant="secondary"
-                        onClick={handleFormSubmit}
-                        disabled={savePerPrioritizationPending
-                            || readOnlyMode}
-                    >
-                        {strings.saveLabel}
-                    </Button>
+                    {value.is_draft === false ? (
```
Comment (tnagorra) [inline src/views/PerPrioritizationForm/index.tsx:420]: What is value.is_draft is undefined? Shouldn't it be treated as false?

File: src/views/DrefApplicationExport/index.tsx [hunk 3]
```diff
@@ -49,6 +49,11 @@ export function Component() {
         onSuccess: () => {
             async function waitForImages() {
                 const images = document.querySelectorAll('img');
+                if (images.length === 0) {
+                    setPreviewReady(true);
+                    return;
+                }
```
Comment (tnagorra) [inline src/views/DrefApplicationExport/index.tsx:56]: What about other exports?

File: src/views/CountryPreparedness/index.tsx [hunk 4]
```diff
@@ -579,6 +571,22 @@ export function Component() {
                     description="Assessment has not been performed yet for current PER cycle!"
                 />
             )}
+            {limitedAccess && (
+                <Message
+                    className={styles.limitedAccessAction}
+                    title="Limited access!"
+                    description="You do not have enough permission to view more details on this PER."
```
Comment (tnagorra) [inline src/views/CountryPreparedness/index.tsx]: Let's use translations

File: src/views/AccountInformation/index.tsx [hunk 5]
```diff
@@ -40,26 +41,33 @@ export function Component() {
...
             offset: ITEM_PER_PAGE * (page - 1),
-            // FIXME: typings should be fixed in server
-        } as never,
+        },
         preserveResponse: true,
     });
 
     const meResponse = useUserMe();
 
+    // FIXME: we do not need this
```
Comment (tnagorra) [inline src/views/AccountInformation/index.tsx:59]: If we do not need this, why do we need this?

File: src/views/AccountDrefApplications/DrefTableActions/DrefShareModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        url: '/api/v2/dref-share/',
+        body: () => ({
+            dref: drefId,
+            users,
+        }),
+        onSuccess: () => {
+            alert.show(
+                'Successfull updated sharing preferences!',
+                { variant: 'success' },
+            );
```
Comment (tnagorra) [inline src/views/AccountDrefApplications/DrefTableActions/DrefShareModal/index.tsx]: Let's use translations

File: src/views/PerWorkPlanForm/index.tsx (hunk 1 above)
Comment (frozenhelium) [reply src/views/PerWorkPlanForm/index.tsx:275]: No, `is_draft` will be undefined when creating the workplan

Comment (frozenhelium) [reply src/views/PerWorkPlanForm/index.tsx:275]: We only hide submit button if the form is already submitted

File: src/views/AccountDrefApplications/DrefTableActions/DrefExportModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+        url: '/api/v2/pdf-export/{id}/',
+        pathVariables: isDefined(exportId) ? ({ id: exportId }) : undefined,
+        onSuccess: (response) => {
+            if (response.status === EXPORT_STATUS_PENDING) {
+                window.clearTimeout(timeoutRef.current);
+                timeoutRef.current = window.setTimeout(() => {
+                    retriggerExportStatus();
+                }, 6000);
+            }
+        },
```
Comment (tnagorra) [inline src/views/AccountDrefApplications/DrefTableActions/DrefExportModal/index.tsx]: We can use another method to do this.
Let's use shouldPoll method

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/PerPrioritizationForm/index.tsx (hunk 2 above)
Comment (frozenhelium) [reply src/views/PerPrioritizationForm/index.tsx:420]: No, is_draft will be undefined when creating the prioritization

**Review by tnagorra — APPROVED** [review]

**Pull request merged**